src/components/feedback/Popover.tsx

```diff
@@ -0,0 +1,260 @@
+    //   right side of viewport
+    // - right-aligned popovers: distance from right side of anchor element to
+    //   left side of viewport
+    const availableSpace =
+      (alignToRight ? anchorElLeft + anchorElWidth : bodyWidth - anchorElLeft) -
+      POPOVER_VIEWPORT_HORIZONTAL_GAP;
+
+    let left = anchorElLeft;
+    if (popoverWidth > availableSpace) {
+      // If the popover is not going to fit the available space, let it "grow"
+      // in the opposite direction
+      left = alignToRight
+        ? POPOVER_VIEWPORT_HORIZONTAL_GAP
+        : left - (popoverWidth - availableSpace);
+    } else if (alignToRight && popoverWidth > anchorElWidth) {
+      // If a right-aligned popover fits the available space, but it's bigger
+      // than the anchor element, move it to the left so that it is aligned with
+      // the right side of the element
+      left -= popoverWidth - anchorElWidth;
+    }
+
+    return setPopoverCSSProps({
+      minWidth: `${anchorElWidth}px`,
+      top: shouldBeAbove
+        ? `calc(${absBodyTop + anchorElDistanceToTop - popoverHeight}px - ${POPOVER_ANCHOR_EL_GAP})`
+        : `calc(${absBodyTop + anchorElDistanceToTop + anchorElHeight}px + ${POPOVER_ANCHOR_EL_GAP})`,
+      left: `${Math.max(POPOVER_VIEWPORT_HORIZONTAL_GAP, left)}px`,
+    });
+  }, [asNativePopover, anchorElementRef, popoverRef, alignToRight]);
+
+  useLayoutEffect(() => {
+    if (!popoverOpen) {
+      return () => {};
+    }
+
+    // First of all, open popover if it's using the native API, otherwise its
+    // size is 0x0 and positioning calculations won't work.
+    const popover = popoverRef.current;
+    if (asNativePopover) {
+      popover?.togglePopover(true);
+    }
+
+    const cleanup = adjustPopoverPositioning();
+
+    if (!asNativePopover) {
+      return cleanup;
+    }
+
+    // Readjust popover position when any element scrolls, just in case that
+    // affected the anchor element position.
+    const listeners = new ListenerCollection();
+    listeners.add(document.body, 'scroll', adjustPopoverPositioning, {
+      capture: true,
+    });
+
+    return () => {
+      if (asNativePopover) {
+        popover?.togglePopover(false);
+      }
+      cleanup();
+      listeners.removeAll();
+    };
+  }, [adjustPopoverPositioning, asNativePopover, popoverOpen, popoverRef]);
+}
+
+export type PopoverProps = {
+  children?: ComponentChildren;
+  classes?: string | string[];
+  variant?: 'panel' | 'custom'; // TODO 'tooltip'
+
+  /** Whether the popover is currently open or not. Defaults to false */
+  open: boolean;
+  /** The element relative to which the popover should be positioned */
+  anchorElementRef: RefObject<HTMLElement | undefined>;
+
+  /**
+   * Determines to what side of the anchor element should the popover be
+   * aligned.
+   * Defaults to `left`
+   */
+  align?: 'right' | 'left';
+
+  /**
+   * Determines if focus should be restored when the popover is closed.
+   * Defaults to false.
```

Comment:
> Can you remind me why this prop defaults to false? I think for most use cases we would probably want it to be true.

Yep, I think the same. I made it `false` by default mainly for two reasons:

a) Consistency with `Dialog` component, where it's false by default.
b) Allow passing `restoreFocusOnClose` rather than `restoreFocusOnClose={false}` when you want to override the default value.

None of them is particularly important, and we have some exceptions to b, because we have agreed not setting the default value for the sake of aesthetics when setting props.

So I would be ok setting it to `true`, or even applying the logic unconditionally and not adding a prop.

Reply:
I finally kept the `restoreFocusOnClose` prop, but made it default to `true`.